Library/src/Systems/Organizations/OrganizationSystem.cpp

```diff
@@ -129,6 +123,139 @@ OrganizationSystem::~OrganizationSystem()
+{
+	csp::common::String SelectedOrganizationId;
+
+	if (OrganizationId.HasValue())
+	{
+		SelectedOrganizationId = *OrganizationId;
+	}
+	else
+	{
+		SelectedOrganizationId = GetCurrentOrganizationId();
+
+		if (SelectedOrganizationId.IsEmpty())
+		{
+			INVOKE_IF_NOT_NULL(Callback, MakeInvalid<OrganizationResult>());
+
+			return;
+		}
+	}
+
+	csp::services::ResponseHandlerPtr ResponseHandler
+		= OrganizationApi->CreateHandler<OrganizationResultCallback, OrganizationResult, void, chs::OrganizationDto>(Callback, nullptr);
+
+	static_cast<chs::OrganizationApi*>(OrganizationApi)->apiV1OrganizationsOrganizationIdGet(SelectedOrganizationId, ResponseHandler);
+}
+
+const csp::common::String& OrganizationSystem::GetCurrentOrganizationId() const
+{
+	// todo: The AuthDto contains an array of Organization Ids but we have no way of knowing which one the user is authenticated against.
+	// Currently people can only belong to a single organization so we can hard code the index. Ticket OF-1291, outlines the work required to
+	// extract this information from the Bearer token claims, where the 'current' Organization Id is also captured.
+	const auto* UserSystem		= csp::systems::SystemsManager::Get().GetUserSystem();
+	const auto& OrganizationIds = UserSystem->GetLoginState().OrganizationIds;
+
+	if (OrganizationIds.Size() == 0)
+	{
+		CSP_LOG_ERROR_MSG("Unable to get current Orgaization Id, you do not belong to an Organization.");
+
+		static const String EmptyId = "";
+		return EmptyId;
+	}
+
+	return OrganizationIds[0];
+}
+
+void OrganizationSystem::CreateOrganization(const csp::common::String& OrganizationOwnerId,
+											const csp::common::String& OrganizationName,
+											OrganizationResultCallback Callback)
+{
+	auto OrganizationInfo = std::make_shared<chs::OrganizationDto>();
+	OrganizationInfo->SetName(OrganizationName);
+	OrganizationInfo->SetOrganizationOwnerId(OrganizationOwnerId);
+
+	csp::services::ResponseHandlerPtr ResponseHandler
+		= OrganizationApi->CreateHandler<OrganizationResultCallback, OrganizationResult, void, chs::OrganizationDto>(Callback, nullptr);
+
+	static_cast<chs::OrganizationApi*>(OrganizationApi)->apiV1OrganizationsPost(OrganizationInfo, ResponseHandler);
+}
+
+void OrganizationSystem::UpdateOrganization(const csp::common::Optional<csp::common::String>& OrganizationId,
+											const csp::common::String& Name,
+											OrganizationResultCallback Callback)
+{
+	if (Name.IsEmpty())
+	{
+		CSP_LOG_WARN_MSG("A valid new Organization name must be specified.");
+
+		INVOKE_IF_NOT_NULL(Callback, MakeInvalid<OrganizationResult>());
+
+		return;
+	}
+
+	csp::common::String SelectedOrganizationId;
+
+	if (OrganizationId.HasValue())
+	{
+		SelectedOrganizationId = *OrganizationId;
+	}
+	else
+	{
+		SelectedOrganizationId = GetCurrentOrganizationId();
+
+		if (SelectedOrganizationId.IsEmpty())
+		{
+			INVOKE_IF_NOT_NULL(Callback, MakeInvalid<OrganizationResult>());
```

Comment:
Could we also print to the log here to tell the user that they attempted to update an organization, but CSP did not do anything, because no ID parameters was given and the user doesn't currently belong to one?

Reply:
Added an error log.